nav2_mppi_controller/src/critics/constraint_critic.cpp

```diff
@@ -50,40 +48,31 @@ void ConstraintCritic::score(CriticData & data)
   auto diff = dynamic_cast<DiffDriveMotionModel *>(data.motion_model.get());
   if (diff != nullptr) {
     if (power_ > 1u) {
-      data.costs += xt::pow(
-        xt::sum(
-          (std::move(
-            xt::maximum(data.state.vx - max_vel_, 0.0f) +
-            xt::maximum(min_vel_ - data.state.vx, 0.0f))) *
-          data.model_dt, {1}, immediate) * weight_, power_);
+      data.costs += (((((data.state.vx - max_vel_).max(0.0f) + (min_vel_ - data.state.vx).
```

Comment:
Question: We have this `power > 1u` thing because in xtensor, theres a difference in performance when even taking `N^1` even though it is just `N`. If Eigen doesn't have that issue, we can remove the annoying if/else in all the critics

Reply:
Ah, Eigen also has this issue. I tried benchmarking after removing if/else, timing dropped from 3.65 ms to 3.90 ms for batch_size = 2000 and time_steps = 50.

Reply:
Cool, OK